Feature/audacity3 podcast chapters json export

nosenfield opened the pull request:
## Title
feat: Add Podcast 2.0 Chapters JSON export for label tracks

## Description
Resolves: https://github.com/audacity/audacity/issues/7078

Add support for exporting label tracks in Podcast 2.0 Chapters JSON format, enabling podcasters to create chapter markers compatible with modern podcast apps.

## Changes

- Add `PODCAST_CHAPTERS_JSON` to `LabelFormat` enum in `LabelTrack.h`
- Add `PodcastChaptersFiles` file type constant in `LabelTrack.cpp`
- Update `FormatForFileName()` to detect `.json` extension
- Implement JSON export in `LabelStruct::Export()` with proper string escaping
- Implement JSON export in `LabelTrack::Export()` with JSON header/footer
- Add "Podcast Chapters" option to Export Labels dialog in `FileMenus.cpp`

<!-- Use "x" to fill the checkboxes below like [x] -->

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

QA:

- [ ] Autobot test cases have been run

nosenfield committed:
feat: implement Podcast 2.0 Chapters JSON export for label tracks

Add support for exporting label tracks in Podcast 2.0 Chapters JSON format.

- Add PODCAST_CHAPTERS_JSON to LabelFormat enum
- Add PodcastChaptersFiles file type constant
- Update FormatForFileName to detect .json extension
- Implement JSON export in LabelStruct::Export with proper string escaping
- Implement JSON export in LabelTrack::Export with header/footer
- Add JSON option to export dialog in FileMenus.cpp

The implementation follows Podcast 2.0 spec with version 1.2.0 and
properly escapes special characters in chapter titles including \b and \f.
Uses 3-space indentation per Audacity coding standards.

Task: Feature 1 - Podcast 2.0 JSON Export (Issue #7078)

nosenfield committed:
fix: update JSON export file type name and extension handling

- Change file type name from 'Podcast Chapters JSON' to 'JSON file' to match existing naming conventions (e.g., 'WebVTT file', 'SubRip text file')
- Ensure filename has .json extension when JSON format is selected in export dialog
- Fixes issue where file extension doesn't update when JSON format is selected

Task: Feature 1 - Podcast 2.0 JSON Export (Issue #7078)

nosenfield committed:
fix: change JSON export display name to 'Podcast Chapters' for better UX

Change file type display name from 'JSON file' to 'Podcast Chapters' to
better communicate the purpose of the export format to users.

Rationale:
- Users want to export podcast chapters, not 'a JSON file'
- The .json extension already indicates the file format
- More user-friendly and purpose-focused
- Matches user mental model and intent

Task: Feature 1 - Podcast 2.0 JSON Export (Issue #7078)

kryksyh commented on libraries/lib-label-track/LabelTrack.cpp:
```diff
@@ -603,6 +604,28 @@ void LabelStruct::Export(wxTextFile &file, LabelFormat format, int index) const
+   {
+      // Individual chapter object (called from LabelTrack::Export)
+      // Format: {"startTime": X.XXX, "title": "..."}
+      // Podcast 2.0 Chapters spec: https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#chapters
```
https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/examples/chapters/jsonChapters.md

kryksyh commented on libraries/lib-label-track/LabelTrack.cpp:
```diff
@@ -679,11 +702,36 @@ void LabelTrack::Export(wxTextFile & f, LabelFormat format) const
+   int numLabels = mLabels.size();
+   for (auto &labelStruct: mLabels) {
+      bool isLast = (index == numLabels - 1);
+      labelStruct.Export(f, format, index);
```
Please add isLast as an argument to LabelStruct::Export, and move the comma logic there.

kryksyh commented on src/menus/FileMenus.cpp:
```diff
@@ -336,6 +336,14 @@ void OnExportLabels(const CommandContext &context)
 
+   // Ensure filename has correct extension for selected format
+   if (format == LabelFormat::PODCAST_CHAPTERS_JSON && fName.Right(5).CmpNoCase(wxT(".json")) != 0) {
+      // Add .json extension if not present
```
The comment here does not match what the code does, it overrides the extension if it does not match ".json", do we really want to restrict the user from exporting the file with an alternative extension?

kryksyh reviewed (changes requested):
@nosenfield thanks for this submission!

nosenfield committed:
refactor: address PR feedback for Podcast 2.0 JSON export

- Fix JSON indentation to 2-space-per-level (matching spec)
- Extract escaping logic into EscapeJSON() helper function
- Move comma handling into LabelStruct::Export() with isLast parameter
- Extract header/footer logic into ExportHeader()/ExportFooter() functions
- Remove forced .json extension (let file dialog handle it)

Addresses PR review feedback from kryksyh

nosenfield committed:
docs: remove misleading comment about JSON escape order
